xlators/mgmt/glusterd/src/glusterd-volgen.c

```diff
@@ -2202,6 +2202,35 @@ brick_graph_add_quota(volgen_graph_t *graph, glusterd_volinfo_t *volinfo,
     return ret;
 }
 
+static int
+brick_graph_add_simple_quota(volgen_graph_t *graph, glusterd_volinfo_t *volinfo,
+                             dict_t *set_dict, glusterd_brickinfo_t *brickinfo)
+{
+    xlator_t *xl = NULL;
+    xlator_t *this = THIS;
+    GF_ASSERT(this);
+
+    if (!graph || !volinfo || !set_dict) {
+        gf_smsg(this->name, GF_LOG_ERROR, errno, GD_MSG_INVALID_ARGUMENT, NULL);
+        goto out;
+    }
+
+    xl = volgen_graph_add(graph, "features/simple-quota", volinfo->volname);
+    if (!xl)
+        goto out;
+
+    /* set no-distribute option in valid cases */
+    int dist_count = volinfo->brick_count / volinfo->dist_leaf_count;
+    if (dist_count > 1)
+        goto out;
+
+    int ret = xlator_set_fixed_option(xl, "no-distribute", "true");
```

Comment:
Is this “no-distribute” option useful， or obsolete？ 
and， no found it in options.

Reply:
this was an optimization to not run any clustering/aggregation of quota sizes if there is no distribute (ie, a single brick, or a 1x3 replica3 volume etc). It is helpful for sure as no quota scripts are required when this option is set.